test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Server/PeopleController.cs

```diff
@@ -0,0 +1,124 @@
+using Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd;
+
+namespace Microsoft.OData.Client.E2E.Tests.ActionOverloadingTests.Server
+{
+    public class PeopleController : ODataController
+    {
+        public PeopleController() { }
+
+        [EnableQuery]
+        [HttpGet("odata/People")]
+        public IActionResult Get()
+        {
+            var people = CommonEndToEndDataSource.People;
+            return Ok(people);
+        }
+
+        [EnableQuery]
+        [HttpGet("odata/People({key})")]
+        public IActionResult Get(int key)
+        {
+            var person = CommonEndToEndDataSource.People?.FirstOrDefault(a => a.PersonId == key);
+            return Ok(person);
+        }
+
+        [HttpPost("odata/Default.UpdatePersonInfo")]
+        public IActionResult UpdatePersonInfo()
+        {
+            CommonEndToEndDataSource.People.First().Name += "[UpdataPersonName]";
+            return Ok();
+        }
+
+        [HttpPost("odata/People({key})/Default.UpdatePersonInfo")]
+        public IActionResult UpdatePersonInfo([FromODataUri] int key)
+        {
+            CommonEndToEndDataSource.People.First(a => a.PersonId == key).Name += "[UpdataPersonName]";
+            return Ok();
+        }
+
+        [HttpPost("odata/People({key})/Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Person/Default.UpdatePersonInfo")]
+        public IActionResult UpdateEmployeeInfo([FromODataUri] int key)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest();
+            }
+
+            CommonEndToEndDataSource.People.First(a => a.PersonId == key).Name += "[UpdataPersonName]";
+
+            return Ok();
+        }
+
+        [HttpPost("odata/People({key})/Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Employee/Default.UpdatePersonInfo")]
+        public IActionResult UpdateEmployeeeInfo([FromODataUri] int key)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest();
+            }
+
+            CommonEndToEndDataSource.People.First(a => a.PersonId == key).Name += "[UpdataPersonName]";
+
+            return Ok();
+        }
+
+        [HttpPost("odata/People({key})/Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Employee/Default.IncreaseEmployeeSalary")]
+        public IActionResult IncreaseEmployeeSalary([FromODataUri] int key, ODataActionParameters parameters)
+        {
+            var person = CommonEndToEndDataSource.People.First(a => a.PersonId == key);
+            if (person is SpecialEmployee specialEmployee && parameters == null)
+            {
+                specialEmployee.Salary += 1;
+                return Ok(specialEmployee.Salary);
+            }
+            else
+            {
+                var employee = (Employee)CommonEndToEndDataSource.People.First(a => a.PersonId == key);
+                employee.Salary += (int)parameters["n"];
+
+                return Ok(true);

```

Comment:
I wonder what the return type of this action is declared in the IEdmModel since here it appears to return different things depending on the type.

Reply:
This is how it is declared: 

```c#
            builder.EntityType<SpecialEmployee>()
                .Action("IncreaseEmployeeSalary")
                .Returns<int>();

            builder.EntityType<Employee>()
                .Action("IncreaseEmployeeSalary")
                .Returns<bool>()
                .Parameter<int>("n");
```

Reply:
For this particular test it is the second one.